packages/data-context/src/sources/RelevantRunSpecsDataSource.ts

```diff
@@ -141,35 +141,27 @@ export class RelevantRunSpecsDataSource {
 
       const { current, next } = cloudProject
 
-      if (
-        current
-         && current.runNumber
-         && current.status
-         && typeof current.totalInstanceCount === 'number'
-         && typeof current.completedInstanceCount === 'number'
-      ) {
-        runSpecsToReturn.runSpecs.current = {
-          totalSpecs: current.totalInstanceCount,
-          completedSpecs: current.completedInstanceCount,
-          runNumber: current.runNumber,
+      const formatCloudRunInfo = (cloudRunDetails: Partial<CloudRun>) => {
+        const { runNumber, totalInstanceCount, completedInstanceCount } = cloudRunDetails
+
+        if (runNumber && Number.isFinite(totalInstanceCount) && Number.isFinite(completedInstanceCount)) {
+          return {
+            totalSpecs: totalInstanceCount!,
+            completedSpecs: completedInstanceCount!,
+            runNumber,
+          }
```

Comment:
I am still wrapping my head around the best way to do type checking in scenarios like this to confirm that variables are defined.  It stinks that you had to use the `!` to tell typescript that these variables have a value since the `Number.isFinite` does not do that for you.  

This is just a comment and not a blocking request.

Reply:
Yuck, hadn't considered the typecheck loss. Could use `_.isNumber` which has an appropriate type guard, but that accepts `NaN` and `*Infinity` which isn't ideal. I think we'd have to build a custom type guard to handle our definition of a valid number to get the TS compiler to handle this automatically:

```
function isValidNumber(value: unknown): value is number {
  return Number.isFinite(value)
}
...
if (runNumber && isValidNumber(totalInstanceCount) && isValidNumber(completedInstanceCount)) {
  return {
    totalSpecs: totalInstanceCount,
    completedSpecs: completedInstanceCount,
    runNumber,
  }
}
```

Reply:
Yeah that seems better than the `!`

Reply:
updated